app.go

```diff
@@ -131,6 +131,18 @@ func (io invokeOption) String() string {
 	return fmt.Sprintf("fx.Invoke(%s)", strings.Join(items, ", "))
 }
 
+// Error registers any number of errors with the application to short-circuit
+// startup. If more than one error is given, the errors are combined into a
+// single error.
+//
+// Similar to invocations, errors are applied in order. All Provide and Invoke
+// options registered before or after an Error option will not be applied.
+func Error(errs ...error) Option {
+	return optionFunc(func(app *App) {
+		app.err = multierr.Append(app.err, multierr.Combine(errs...))
```

Comment:
When `Error` is applied, does it end up stopping the rest of the constructors/invokes? Want to make sure we don't see errors about missing types (because there was some other error instead)

can we verify that through unit tests for both `Provide` and `Invoke` relying on a missing type

Reply:
As the comment describes, all `Provide` and `Invoke` options registered before/after an `Error` option will not be applied. The `SingleErrorOption` and `MultipleErrorOption` sub-tests in `TestError` both test this behavior. 

I will add tests focused around a `Provide` and `Invoke` missing dependencies to further verify this.